[config template] Document log_to_console and syslog_* options

olivielpeau opened the pull request:
### What does this PR do?

Documents `log_to_console` and `syslog_*` options in the example yaml file.

### Motivation

These were previously not documented (for no good reason I think).

### Additional notes

While documenting this, I realized that `syslog_pem` was expecting a raw string of PEM encoded certificates. Would it make more sense to allow specifying a path instead? cc @truthbk 

update: I've left this option undocumented for now, we'll open a separate PR to change its behavior

olivielpeau committed:
[config template] Document log_to_console and syslog_* options

These were previously not documented (for no good reason I think).

arbll commented on pkg/config/config_template.yaml:
```diff
@@ -241,6 +238,33 @@ api_key:
 # log_file: /var/log/datadog/agent.log
+
+# Set to "yes" to output logs in JSON format
+# log_format_json: no
```
Shouldn't this be `false` instead of `no` ?

arbll reviewed (commented):
LGTM, just a question

olivielpeau committed:
Add disable_file_logging

truthbk commented on pkg/config/config_template.yaml:
```diff
@@ -241,6 +238,36 @@ api_key:
+#
+# If 'syslog_tls' is enabled, you can set your pem-encoded root ca certs here
+#
+# syslog_pem:
```
you're 100% right... this actually sucks 🙈  
These should probably be paths to PEM certificates, and we'll then have to load those into memory (into `x509.Certificate`s and append those to the logger). I can fix the implementation... Can I push to this branch?

truthbk reviewed (commented):
Yeah, this sucks in its current state. My bad.

olivielpeau committed:
Remove docs on syslog_pem option

We'll want to change what the option takes as value soon. For now let's
not document it

truthbk reviewed (approved):
Looks good, thanks for removing that 👌
